File: angular-skiecommerce/src/app/classes/product.ts
import { ProductCategory } from "./product-category";

export class Product {

    id!: number;
    brand!: string;
    name!: string;
    size!: number;
    condition!: string;
    unitPrice!: number;
    unitsInStock!: number; 
    special!: string;
    imageUrl!: string;
    productKey!: string;
    cateogry!: ProductCategory
}

File: skiecommenrce-backed/skiecommenrce-backed/src/main/java/pl/opalka/skiecommenrcebacked/model/Product.java
package pl.opalka.skiecommenrcebacked.model;

import lombok.Data;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Data
public class Product {

    @Id
    private Long id;

    private String brand;

    private String name;

    private String condition;

    private BigDecimal unitPrice;

    private int unitsInStock;

    private String special;

    private String imageUrl;

    private double size;

    private String productKey;

    @ManyToOne()
    @JoinColumn(name = "category_id")
    private ProductCategory category;
}
